lib/IMAP/IMAPClientFactory.php

```diff
@@ -117,8 +117,16 @@ public function getClient(Account $account, bool $useCache = true): Horde_Imap_C
 				'backend' => $this->hordeCacheFactory->newCache($account),
 			];
 		}
-		if ($this->config->getSystemValue('debug', false)) {
-			$params['debug'] = $this->config->getSystemValue('datadirectory') . '/horde_imap.log';
+		$debug = array_merge(
+			explode('|', $this->config->getSystemValueString('MAIL_DEBUG')),
```

Comment:
I'm aware that getSystemValueString is used here as a workaround for reading environment variables, but I have some concerns:

- Although we don't have strict conventions for naming configuration variables, all variables in [config.sample.php](https://github.com/nextcloud/server/blob/master/config/config.sample.php) use lowercase. I’d prefer to keep this consistent.
- For configuration options related to SMTP in [config.sample.php](https://github.com/nextcloud/server/blob/master/config/config.sample.php), we currently use the mail_ prefix. While smtp_ might be more accurate, it’s not feasible to change this now :disappointed: 
- Our standard format is usually app.mail.xyz, but this could be problematic when working with NC_ environment variables due to naming restrictions in Linux (see [allowed characters in Linux environment variable names](https://stackoverflow.com/questions/2821043/allowed-characters-in-linux-environment-variable-names)).

Reply:
- 1 sure I can change it to lower case

- 2 yes I could make it support mail_ and smtp_ or anything else and this would enable global debugging, but the idea behind this PR, is to NOT touch the config file at all for mail communication issues, also to NOT enable debugging for all user(s) mail account(s) at the same time and to separate the transmission logs so that its easier and faster to find issues. Just imagine if you have a system with 100 mail accounts and only 2 accounts have an intermittent problem a few times a day. 
yes I know that by adding the 'getSystemValueString' you can now set the debugging in the config.php, but the purpose was for one time environmental variables.

- 3 yes that is why it has to stay with an under sore, the only other option is one word

Reply:
> For configuration options related to SMTP in [config.sample.php](https://github.com/nextcloud/server/blob/master/config/config.sample.php), we currently use the mail_ prefix. While smtp_ might be more accurate, it’s not feasible to change this now 😞

I was referring to the symfony mailer configuration in Nextcloud using `mail_` as prefix and thus consider mail_debug not an ideal choice.

Reply:
> yes I know that by adding the 'getSystemValueString' you can now set the debugging in the config.php, but the purpose was for one time environmental variables.

In a previous version, you were using $_ENV / getenv, why did you withdraw this approach? I think that would be cleaner than hijacking SystemConfig :pirate_flag:

Reply:
> I was referring to the symfony mailer configuration in Nextcloud using `mail_` as prefix and thus consider mail_debug not an ideal choice.

Ah, Okay, yeah we can still change it to mail_app_debug or anything else.

> In a previous version, you were using $_ENV / getenv, why did you withdraw this approach? I think that would be cleaner than hijacking SystemConfig 🏴‍☠️

I agree it was cleaner but abandoned the approach, ironically because I thought it would get flagged during the review process, for one of two reasons,

- A. Because I was overriding account data in the MailAccount class with information from a different source. Didn't think this was best practice.
- B. Because I was using raw system values